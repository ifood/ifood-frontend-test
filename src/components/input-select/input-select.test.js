import React from 'react'
import { render } from '@testing-library/react'
import { InputSelect } from '.'

describe('InputSelect', () => {
  let component

  beforeEach(() => {
    const props = {
      options: [
        {
          name: 'Option 1',
          value: '1'
        },
        {
          name: 'Option 2',
          value: '2'
        }
      ],
      id: '1',
      name: 'country',
      label: 'Country',
      value: '1',
      onChange: jest.fn(x => x)
    }

    component = render(<InputSelect {...props} />)
  })

  it('should render component correctly', () => {
    const { asFragment } = component
    expect(asFragment()).toMatchSnapshot()
  })
})
