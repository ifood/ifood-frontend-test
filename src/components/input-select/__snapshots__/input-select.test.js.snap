// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSelect should render component correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root makeStyles-formControl-1"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
    >
      Country
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="1 1"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="1"
        role="button"
        tabindex="0"
      >
        Option 1
      </div>
      <input
        type="hidden"
        value="1"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
